### ISSUE TITLE
TOTP throw error RainLoop\Exceptions\ClientException: AuthError[102]

### ISSUE THREAD

**Issue by Nono-m0le** [the-djmaze/snappymail#1283]
**Describe the bug**
With snappymail 2.29.1 & Two Factor Authentication 2.16.5, when the 2FA is set on an account it's not possible to log in, it's returning to the auth page again and again.

**To Reproduce**
1. Enable & setup 2FA, test it with success
2. Try to login using a (working) 2FA code
3. It never login, but return to the login page

**Expected behavior**
We should be able to login ... on this case : no error is shown on the gui.

**Please complete the following information:**
- Browser: Chrome & Firefox, latest version
- IMAP daemon: dovecot
- PHP version: 8.2.7
- SnappyMail Version: 2.29.1
- Mode: standalone

**[Debug/logging information](https://github.com/the-djmaze/snappymail/wiki/FAQ#how-do-i-enable-logging)**
```
[2023-10-12 13:54:18.280][f6b38a83] [ERROR]: RainLoop\Exceptions\ClientException: AuthError[102] in /path/to/snappy/data/_data_/_default_/plugins/two-factor-auth/index.php:71
Stack trace:
#0 /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/Plugins/Manager.php(314): TwoFactorAuthPlugin->AfterImapLogin()
#1 /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/Model/Account.php(299): RainLoop\Plugins\Manager->RunHook()
#2 /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/Model/Account.php(238): RainLoop\Model\Account->netClientLogin()
#3 /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/Actions.php(1040): RainLoop\Model\Account->ImapConnectAndLogin()
#4 /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/Actions/Folders.php(50): RainLoop\Actions->initMailClientConnection()
#5 /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/ServiceActions.php(142): RainLoop\Actions->DoFolders()
#6 /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/Service.php(132): RainLoop\ServiceActions->ServiceJson()
#7 /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/Service.php(14): RainLoop\Service::RunResult()
#8 /path/to/snappy/snappymail/v/2.29.1/include.php(143): RainLoop\Service::Handle()
#9 /path/to/snappy/index.php(11): include('...')
#10 {main}

Next RainLoop\Exceptions\ClientException: AuthError[102] in /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/Actions.php:1044
Stack trace:
#0 /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/Actions/Folders.php(50): RainLoop\Actions->initMailClientConnection()
#1 /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/ServiceActions.php(142): RainLoop\Actions->DoFolders()
#2 /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/Service.php(132): RainLoop\ServiceActions->ServiceJson()
#3 /path/to/snappy/snappymail/v/2.29.1/app/libraries/RainLoop/Service.php(14): RainLoop\Service::RunResult()
#4 /path/to/snappy/snappymail/v/2.29.1/include.php(143): RainLoop\Service::Handle()
#5 /path/to/snappy/index.php(11): include('...')
#6 {main}
```

**Additional context**
This is the following of another (closed) ticket : https://github.com/the-djmaze/snappymail/issues/1241

**Comment by bob912** [comment]
I am also having this issue trying to enable TOTP codes - Test worked fine in Settings screen but then login just keeps returning to authentication screen. Have disabled TOTP in admin page and can log in again as a user.

**Comment by tinola** [comment]
I can confirm issue. Same here.

**Comment by the-djmaze** [comment]
Reverted the plugin code and changed the app code to solve this

**Comment by apezio** [comment]
Not sure if this is the same issue, but I had a problem where after entering my correct login and 2fa token, I would be redirected to the login screen again.

It was related to my login username for snappymail, and username in the two-factor filenot matching.  I bypassed the problem by editing "User" in:

/var/lib/snappymail/_data_/_default_/storage/<domain.tld>/<username>/two_factor

So that instead of having my email address as username, I changed it to just my username.
